xtask/src/docgen.rs

```diff
@@ -95,14 +94,14 @@ pub fn lang_features() -> Result<String, DynError> {
                 .to_owned(),
             );
         }
-        row.push(
-            lc.language_servers
-                .iter()
-                .filter_map(|ls| config.language_server.get(&ls.name))
-                .map(|s| md_mono(&s.command.clone()))
-                .collect::<Vec<_>>()
-                .join(", "),
-        );
+        let mut v: Vec<String> = lc.language_servers
+                            .iter()
+                            .filter_map(|ls| config.language_server.get(&ls.name))
+                            .map(|s| md_mono(&s.command.clone()))
+                            .collect::<Vec<_>>();
+        v.sort();
```

Comment:
I would prefer that we keep the order that we have in the `languages.toml` file for this generated doc since definition order is somewhat important (for example code actions and symbol pickers are ordered by them).

We can use a hash set that we insert names into as we go so that we keep the order and remove duplicates:

```rust
let mut seen_commands = HashSet::new();
let mut commands = String::new();
for ls_config in lc
    .language_servers
    .iter()
    .filter_map(|ls| config.language_server.get(&ls.name))
{
    let command = &ls_config.command;
    if !seen_commands.insert(command) {
        continue;
    }

    if !commands.is_empty() {
        commands.push_str(", ");
    }

    commands.push_str(&md_mono(command));
}
row.push(commands);
```

Reply:
This is my first RUST Foss change contribution.. If you want me to redo it again to use HashMap instead I can.. I was looking for the most simple approach.

Reply:
You should be able to copy that code in and it should do the job. The `HashSet` code.

Reply:
ok I implemented your code using hashset. xtask docgen worked..